tamr_toolbox/notifications/emails.py

```diff
@@ -0,0 +1,206 @@
+"""Tasks related to creation of Email notifications"""
+import logging
+import smtplib
+import ssl
+
+from typing import Union, List, Optional
+from email.mime.text import MIMEText
+from smtplib import SMTPException
+from tamr_toolbox.notifications.common import monitor_job as monitor_job_common
+
+from tamr_unify_client import Client
+from tamr_unify_client.operation import Operation
+
+from tamr_toolbox.models.operation_state import OperationState
+from tamr_toolbox.utils.operation import get_details
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _build_message(
+    *, message: str, subject_line: str, sender: str, recipients: List[str]
+) -> MIMEText:
+    """Builds email message in Multipurpose Internet Mail Extensions (MIME) format. MIME is an
+    Internet standard that extends the format of email messages
+
+    Args:
+        message: Body of email message
+        subject_line: subject of email
+        sender: email address of sender
+        recipients: list of emails to send message to
+
+    Returns:
+        Email as a string
+    """
+
+    # build email
+    msg = MIMEText(message)
+    msg["Subject"] = subject_line
+    msg["From"] = sender
+    msg["To"] = ", ".join(recipients)
+    return msg.as_string()
+
+
+def send_email(
+    *,
+    message: str,
+    subject_line: str,
+    sender_address: str,
+    sender_password: str,
+    recipient_addresses: List[str],
+    smtp_server: str,
+    smtp_port: str,
+    raise_error: bool = True,
+    use_tls: bool = False,
+) -> dict:
+    """Sends a message via email to list of recipients
+
+    Args:
+        message: Body of email message
+        subject_line: subject of email
+        sender_address: email address to send message from ex: my_pipeline@gmail.com
+        sender_password: password to login to sender_email
+        recipient_addresses: list of emails to send message to ex: [client_email@gmail.com]
+        smtp_server: smtp server address of sender_email ex: smtp.gmail.com
+        smtp_port: port to send email from, use 465 for SSL
```

Comment:
Add the optional variables `keyfile` and `certfile` with a default of `None` to be passed to the SMTP_SSL function. This way a user could specify these if they want to authenticate at the client-side and not just on the server side. See https://stackoverflow.com/questions/24250020/sending-email-without-keyfile-only-certfile-using-python-smtplib